pkgs/t3/default.nix

```diff
@@ -0,0 +1,29 @@
+{
+  help2man,
+  lib,
+  stdenv,
+  fetchFromGitHub,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "t3";
+  version = "1.0.3";
+
+  src = fetchFromGitHub {
+    owner = "flox";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-NB1aNnMnFhv5rvTvml7vLSJlH94OW9zplttk8HrAy+8=";
+  };
```

Comment:
nit: slight preference to pull this in via the flake, all other external dependencies are being managed through the input interfaces

Reply:
I strongly prefer moving it to our flake so that flake bumps bump t3 rather than having to go bump this hash manually

Reply:
Thanks Yannik & Matthew - I misunderstood both of your initial suggestions. Talking with Matthew he explained he was looking for something more like the following:
```diff
diff --git a/flake.nix b/flake.nix
index 09fe176c..8dc9fe6f 100644
--- a/flake.nix
+++ b/flake.nix
@@ -27,6 +27,9 @@
   inputs.fenix.url = "github:nix-community/fenix";
   inputs.fenix.inputs.nixpkgs.follows = "nixpkgs";
 
+  inputs.t3-src.url = "github:flox/t3";
+  inputs.t3-src.flake = false;
+
   # -------------------------------------------------------------------------- #
 
   outputs =
@@ -57,7 +60,7 @@
           nix = final.callPackage ./pkgs/nix { };
 
           cpp-semver = final.callPackage ./pkgs/cpp-semver { };
-          t3 = final.callPackage ./pkgs/t3 { };
+          t3 = final.callPackage ./pkgs/t3 { inherit (inputs) t3-src; };
         })
         inputs.sqlite3pp.overlays.default
         inputs.fenix.overlays.default
diff --git a/pkgs/t3/default.nix b/pkgs/t3/default.nix
index fed07c38..9abecacb 100644
--- a/pkgs/t3/default.nix
+++ b/pkgs/t3/default.nix
@@ -3,18 +3,14 @@
   lib,
   stdenv,
   fetchFromGitHub,
+  t3-src,
 }:
 
 stdenv.mkDerivation rec {
   pname = "t3";
   version = "1.0.3";
 
-  src = fetchFromGitHub {
-    owner = "flox";
-    repo = pname;
-    rev = "v${version}";
-    hash = "sha256-NB1aNnMnFhv5rvTvml7vLSJlH94OW9zplttk8HrAy+8=";
-  };
+  src = t3-src;
 
   installFlags = [ "PREFIX=$(out)" ];
   nativeBuildInputs = [ help2man ];
```

I have rebased the above patch into the PR and it works just fine. Resolving conversation.  :+1: